src/handlers/mod.rs

```diff
@@ -707,6 +707,11 @@ impl GammaControlHandler for State {
 }
 delegate_gamma_control!(State);
 
+pub enum ActivationContext {
+    UrgentOnly,
+    FocusIfPossible,
+}
```

Comment:
I'm thinking, maybe leave just a marker `struct UrgentOnlyMarker;`? Since we already filter tokens as valid/invalid by returning true/false from `token_created()`. Then only insert `UrgentOnlyMarker` in the urgent-only branch, and check if it's present in `request_activation()`.